### PR TITLE
Load project's PostCSS config if none is specified

### LINKED ISSUE #31: Load PostCSS config by default
I suggest loading PostCCS config file by default, e.g. `postcss.config.js`, unless a configuration is specified. It makes sense that you'd want the same configuration that you're using for external CSS files.

### PR DESCRIPTION
**Pull request by silvenon**
This results in better PostCSS syntax error messages because they would show the HTML file name.

I was thinking what would be the best way to do this. Currently it looks for the PostCSS config if no arguments are passed, and if it doesn't find it, it doesn't fail. Failing seemed kind of obtrusive to me, but then again, in that case `posthtml-postcss` wouldn't be doing anything, which is unexpected. Maybe I should print out a warning, what do you think?

Fixes #31.

### CODE REVIEW

**Comment by silvenon** [comment]
I'll work on fixing the coverage.

**Comment by Scrum** [comment]
@silvenon Hi

> Maybe I should print out a warning, what do you think?

I think it is worth giving a warning - this is an understanding of the reasons for inoperability

File: index.js [hunk 1]
```diff
@@ -1,6 +1,13 @@
 var postcss = require('postcss')
+var postcssrc = require('postcss-load-config')
 
 module.exports = function (plugins, options, filterType) {
+  if (Array.from(arguments).length === 0) {
```
Comment (Scrum) [inline index.js:5]: the `arguments` has the property `length` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments/length

File: index.js [hunk 2]
```diff
@@ -1,6 +1,13 @@
 var postcss = require('postcss')
+var postcssrc = require('postcss-load-config')
```
Comment (Scrum) [inline index.js:2]: The current package supports `nodejs 8` and higher, we can safely use `const` he is holding on `nodejs` 6 and up

Comment (silvenon) [reply index.js:2]: I didn't want to pollute the diff by changing `var`s to `const`s etc.

Comment (Scrum) [reply index.js:2]: I understand, don't worry about it)

**Comment by silvenon** [comment]
I decided that it's better that `posthtml-postcss` throws if it doesn't find the PostCSS configuration, to match how other PostCSS tools behave. If people don't want any configuration, they can pass an empty array explicitly.

I added another test for checking whether the syntax error message includes the file name, but I need to figure out how to test whether `posthtml-postcss` will actually correctly pick up and use the PostCSS configuration file. I'll try by mocking the `fs` module with [`memfs`](https://github.com/streamich/memfs). Hopefully that will improve the coverage.

File: index.js (hunk 1 above)
Comment (silvenon) [reply index.js]: Thanks, fixed.

**Comment by PeteSolid** [comment]
Any updates on this?

**Comment by Scrum** [comment]
> Any updates on this?

Yes, there is an approval for this PR, it remains to resolve the conflicts, unfortunately I am not able to resolve them. for me the button is disabled.

@silvenon will you have the opportunity to resolve conflicts?

**Comment by silvenon** [comment]
Resolved conflicts. I tried implementing memfs, but I don't know how to mock modules outside of Jest. I tried tools like mock-require and proxyquire, but they don't seem to do what I want. 😕

This is ready for merging. I just want to add an unrelated note that this repo is configured to run ESLint on `pre-commit`, but ESLint is not installed, nor is there an ESLint config, so that should be fixed accordion to the intention (probably `xo`, as that's installed).

**Comment by Scrum** [comment]
Thanks for the work done, it looks very good.

> This is ready for merging. I just want to add an unrelated note that this repo is configured to run ESLint on `pre-commit`, but ESLint is not installed, nor is there an ESLint config, so that should be fixed according to the intention (probably `xo`, as that's installed).

thanks, I have planned this fix #38

**Pull request merged**